sklearn/preprocessing/tests/test_data.py

```diff
@@ -851,6 +853,118 @@ def test_robust_scaler_iris_quantiles():
     assert_array_almost_equal(q_range, 1)
 
 
+def test_quantile_normalizer_iris():
+    X = iris.data
+    normalizer = QuantileNormalizer()
+    X_trans = normalizer.fit_transform(X)
+    # FIXME: one of those will drive to precision error
+    # in the interpolation
+    # assert_array_almost_equal(np.min(X_trans, axis=0), 0.)
+    # assert_array_almost_equal(np.max(X_trans, axis=0), 1.)
+    X_trans_inv = normalizer.inverse_transform(X_trans)
+    assert_array_almost_equal(X, X_trans_inv)
```

Comment:
There are no problem for that case (and it's a way to be sure the normalizer works in a correct way).
But what you say is true indeed for general cases.